usual/tls/tls.c

```diff
@@ -742,4 +742,85 @@ tls_close(struct tls *ctx)
 	return (rv);
 }
 
+static bool
+tls_mems_equal(char *mem1, char *mem2, size_t len1, size_t len2)
+{
+	if (len1 != len2)
+		return false;
+	if (mem1 && mem2 && memcmp(mem1, mem2, len1) != 0)
+		return false;
+	return true;
+}
+
+static bool
+keypair_equal(struct tls_keypair *tkp1, struct tls_keypair *tkp2)
+{
+	if (!strcmpeq(tkp1->cert_file, tkp2->cert_file))
+		return false;
+	if (!tls_mems_equal(tkp1->cert_mem, tkp2->cert_mem, tkp1->cert_len, tkp2->cert_len))
+		return false;
+	if (!strcmpeq(tkp1->key_file, tkp2->key_file))
+		return false;
+	if (!tls_mems_equal(tkp1->key_mem, tkp2->key_mem, tkp1->key_len, tkp2->key_len))
+		return false;
+	return true;
+}
+
+bool
+keypair_list_equal(struct tls_keypair *tkp1, struct tls_keypair *tkp2)
+{
+	bool kp_unchanged;
+	struct tls_keypair *tls_kp_lt, *tls_kp_rt;
+
+	kp_unchanged = keypair_equal(tkp1, tkp2);
+
+	if (kp_unchanged == false) {
+		return false;
+	}
```

Comment:
This seems unnecessary now the same check is done again below.

Reply:
Agreed [5467377](https://github.com/libusual/libusual/pull/57/commits/5467377ef1c17263ebc385e681ccd2addc5ca308)